backend/routes/sse.js

```diff
@@ -0,0 +1,29 @@
+const clients = new Map();
+
+const setupSSE = (app) => {
+  app.get('/events', (req, res) => {
+    res.setHeader('Content-Type', 'text/event-stream');
+    res.setHeader('Cache-Control', 'no-cache');
+    res.setHeader('Connection', 'keep-alive');
+
+    const id = Date.now();
+    clients.set(id, res);
+
+    req.on('close', () => {
+      clients.delete(id);
+    });
+  });
+};
+
+const sendToClients = (message) => {
+  clients.forEach((client, id) => {
```

Comment:
are you sending the event to ALL clients? Where are we filtering to ensure the events are only being sent to the desired client attached to the recipient userId?